app/code/Vindi/Payment/Model/Payment/Vindi.php

```diff
@@ -155,130 +169,109 @@ public function assignData(DataObject $data)
+        if (!array_key_exists(PaymentMethod::$cCBrands[$info->getCcType()], $availableTypes)) {
+            return $this->addError(__('Credit card type is not allowed for this payment method.'));
         }
 
-        $this->getInfoInstance()->setAdditionalInformation(
-            [
-                'installments' => $installments,
-                'nsu' => $nsu
-            ]
-        );
-
-        $stateObject->setStatus(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT)
-            ->setState(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT);
-
         return $this;
     }
 
-    /**
-     * @param int $customerId
-     *
-     * @return array|bool
-     */
-    protected function createPaymentProfile($customerId)
+    protected function validateExpDate($expYear, $expMonth)
     {
-        $payment = $this->getInfoInstance();
-
-        $creditCardData = [
-            'holder_name' => $payment->getCcOwner(),
-            'card_expiration' => str_pad($payment->getCcExpMonth(), 2, '0', STR_PAD_LEFT)
-                . '/' . $payment->getCcExpYear(),
-            'card_number' => $payment->getCcNumber(),
-            'card_cvv' => $payment->getCcCid() ?: '000',
-            'customer_id' => $customerId,
-            'payment_company_code' => $payment->getCcType(),
-            'payment_method_code' => $this->getPaymentMethodCode()
-        ];
-
-        $paymentProfile = $this->api()->createCustomerPaymentProfile($creditCardData);
-
-        if ($paymentProfile === false) {
-            Mage::throwException('Erro ao informar os dados de cartão de crédito. Verifique os dados e tente novamente!');
-
-            return false;
+        $date = $this->date->date();
+        if ($expYear && $expMonth && ($expYear > $date->format('Y'))
+            || ($date->format('Y') == $expYear && ($expMonth > $date->format('m')))
+        ) {
+            return true;
         }
-
-        $verifyMethod = Mage::getStoreConfig('vindi_subscription/general/verify_method');
-
-        if ($verifyMethod && !$this->verifyPaymentProfile($paymentProfile['payment_profile']['id'])) {
-            Mage::throwException('Não foi possível realizar a verificação do seu cartão de crédito!');
-            return false;
-        }
-        return $paymentProfile;
+        return false;
     }
 
-    /**
-     * @param int $paymentProfileId
-     *
-     * @return array|bool
-     */
-    public function verifyPaymentProfile($paymentProfileId)
+    public function capture(\Magento\Payment\Model\InfoInterface $payment, $amount)
     {
-        $verify_status = $this->api()->verifyCustomerPaymentProfile($paymentProfileId);
-        return ($verify_status['transaction']['status'] === 'success');
-    }
+        $order = $payment->getOrder();
+        $customerId = $this->customer->findOrCreate($order);
+        $paymentProfile = $this->profile->create($payment, $customerId, $this->getPaymentMethodCode());
+        $productList = $this->product->findOrCreateProducts($order);
```

Comment:
`$shippingId = $this->findOrCreateProduct('Frete');`
`$productList => [
'product_id'     => $shippingId,
'amount' => $order->getShippingAmount()
];`

Reply:
acredito que isto deva estar dentro da função findOrCreateProducts

Reply:
Ficou bom da maneira que você fez [commit](https://github.com/vindi/vindi-magento2/pull/6/commits/6062b9e98cc44acb05bf5e1785825ccd28e00346) :smiley: